app/bundles/ChannelBundle/Event/ChannelEvent.php

```diff
@@ -71,8 +71,10 @@ public function getRepositoryName($channel)
 
         // if not defined, try the classic naming convention
         $channel = ucfirst($channel);
+        $class   = "\Mautic\\{$channel}Bundle\Entity\\{$channel}";
```

Comment:
@escopecz is this a good approach, or do you have another one?

Reply:
Yep, let's go with this for now.